enumerables.rb

```diff
@@ -1,88 +1,100 @@
+# frozen_string_literal: true
+
+# enumerable methods
 module Enumerable
   def my_each
     arr = self
+    if block_given?
       for i in arr
         yield(i)
+        puts i
       end
+    else
+      arr.to_enum(:my_each)
+    end
   end
 
   def my_each_with_index
     arr = self
-      for i in arr
+    if block_give?
+      arr.my_each do |i|
         v = arr.find_index(i)
         yield(i, v)
       end
+    else
+      arr.to_enum(:my_each_with_index)
+    end
   end
 
   def my_select
     arr = self
     result = []
-    for i in arr
-      if  yield i
-        result.push(i)
-        end
+    arr.each do |i|
+      result.push(i) if yield i
     end
     result
   end
 
-  def my_all(argument = nil)
+  def my_all?(arg = nil)
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false unless yield(i)
+        return false unless yield(i)
       end
     else
-      my_each { |i| return false if i.nil? || i == false }
+      my_each { |i| return false if !i.is_a? arg }
     end
     true
   end
-
-  def my_all(argument = nil)
+  def my_any?( arg = nil )
     arr = self
     if block_given?
-      arr.my_each do |i|
-      return false
+      arr.my_each do |i| 
+        return true if yield(i)
       end
     else
-      my_each { |i| yield i if i.nil? || i == false }
+      my_each { |i| return true if i.is_a? arg }
     end
-    true
+    false
   end
 
-  def my_none(argument = nil)
+  def my_none?( arg = nil )
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false if yield i
+        return false if yield i
       end
     else
-      my_each { |i| yield i if i.nil? && i == false }
+      my_each { |i| return false if i.is_a? arg }
     end
     true
   end
 
   def my_map
     arr = self
-    mapped_arr =[]
-      arr.my_each do |i| i
-        mapped_arr.push(yield i)
-        yield i
-      end
-      mapped_arr
+    mapped_arr = []
+    arr.my_each do |i|
+      mapped_arr.push(yield i)
+      yield i
+    end
+    mapped_arr
   end
 
-  def my_inject
+  def my_inject(result = nil)
     arr = self
-    result = 0
     arr.my_each do |i|
       result = yield(result, i)
-      end
+    end
     result
   end
```

Comment:
* [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
  ```ruby
    array = [4,7,8,89,9,6,5,12,4,2,5]
    operation = proc { |sum, n| sum + n  }
    p array.my_inject(&operation) == array.inject(&operation) # false
  ```
* [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
  ```ruby
    array = [1,7,8,89,9,6,5,12,4,2,5]
    p array.my_inject(:+) == array.inject(:+) # return this error
    # LocalJumpError:
        no block given (yield)
      # ./enumerables.rb:86:in `block in my_inject'
      # ./enumerables.rb:9:in `block in my_each'
      # ./enumerables.rb:8:in `each'
      # ./enumerables.rb:8:in `my_each'
      # ./enumerables.rb:85:in `my_inject'
  ```